Sources/Workspace/WorkspaceState.swift

```diff
@@ -60,45 +42,320 @@ public final class WorkspaceState: SimplePersistanceProtocol {
         // these errors will also be ignored but will surface when clients try
         // to save the state.
         do {
-            try self.persistence.restoreState(self)
+            let storedState = try self.storage.load()
+            self.dependencies = storedState.dependencies
+            self.artifacts = storedState.artifacts
         } catch {
+            self.dependencies = ManagedDependencies()
+            self.artifacts = ManagedArtifacts()
+            try? self.storage.reset()
             // FIXME: We should emit a warning here using the diagnostic engine.
-            TSCBasic.stderrStream.write("\(error)\n")
+            TSCBasic.stderrStream.write("warning: unable to restore workspace state: \(error)")
```

Comment:
I realize this is existing code, so doesn't need to be changed in this PR, but should a future PR change this to use a `Diagnostics` collector since I assume this could be emitted from an IDE as well (as it's part of libSwiftPM)?

Reply:
the issue is that Workspace does not have a DiagnosticsEngine to use in this case. I was planning to address this in one of the next refactoring PRs, focused on making Diagnostics more globally available

Reply:
Yah, we have at least one more such case that we can clean up once we make diagnostics more broadly available.